mod/common/src/main/java/gjum/minecraft/mapsync/common/DimensionChunkMeta.java

```diff
@@ -34,11 +36,19 @@ public class DimensionChunkMeta {
 	}
 
 	private Path getRegionFilePath(RegionPos regionPos) {
-		return Path.of(dimensionDirPath, "r%d,%d.chunkmeta".formatted(regionPos.x, regionPos.z));
+		return Path.of(dimensionDirPath, "r%d,%d.chunkmeta".formatted(regionPos.x(), regionPos.z()));
 	}
 
-	private Path getTsFilePath() {
-		return Path.of(dimensionDirPath, "last_ts");
+	public synchronized boolean requiresChunksFrom(RegionPos regionPos, long latestUpdateTimestamp) {
+		long[] chunkTimestamps = regionsTimestamps.computeIfAbsent(regionPos, this::readRegionTimestampsFile);
+		long newestChunkUpdate = 0;
```

Comment:
we should compare the oldest clientside chunk ts with the newest serverside chunk ts, right? to catch regions that are only partially updated, instead of just comparing the newest chunks on both sides